### DISCUSSION TITLE
As the damage index increases, indicating more damage, the effective stiffness of each spring element in the creep model decreases?

### DISCUSSION THREAD

**Discussion by deyantian18** [idaholab/moose#26070] [Q&A Modules: Solid mechanics] [answered]
Dear Community Members,

I have developed a model that incorporates elasticity, creep model by KV, and damage evolution calculated by energy, and I am interested in understanding whether the evolution of damage affects the creep parameters (e.g. weaken the spring's stiffness in the viscoelasticity model). as my understanding, when the material is in a inleastic-deformed status, the stress is only generated by the elastic strain (σ = C * (ε_total - ε_ine - ε_eigen) = C * ε_e) . But when a damage evolution model is added, the damage index is only affect on the elasticity to **calculate the stress** or affect both the elasticity and elasticity (the creep modulus ) in creep?

here is the input file for material:
[./burgers] 
    type = GeneralizedKelvinVoigtModel
    creep_modulus = '11000 20000'
    creep_viscosity = '15 6000'
    poisson_ratio = 0.2
    young_modulus = 11000
    driving_eigenstrain = ini_stress
[../]
[./strain]
    type = ComputeIncrementalSmallStrain
    displacements = 'disp_x disp_y disp_z'
    eigenstrain_names = ini_stress 
[../]

[./stress]
    type = ComputeMultipleInelasticStress
    inelastic_models = 'creep'
    perform_finite_strain_rotations = false
    tangent_operator='nonlinear'
    damage_model= damage
[../]
[./creep]
    type = LinearViscoelasticStressUpdate
[../]
 [./damage]
    type = Tian_AccuDamageByEnergy
              -
              -
              --
  [../]

many thanks.

**Comment by bwspenc** [comment]
The damage model only has the effect of decreasing the stress computed from the elastic strain, so the damaged stress calculation would look like: σ = (1-d) * C * (ε_total - ε_ine - ε_eigen) = (1-d) * C * ε_e
Also, it's important to note that at the beginning of the inelastic calculation, we undo the application of damage on the stress from the previous step, so we go through the whole calculation of creep as if there were no damage. Since in the end the stress gets scaled down due to damage, the elastic stiffness as well as the stiffnesses of the springs in the creep model effectively get decreased by the same amount.

**Reply by bwspenc** [reply]
There are only a couple of places in the code where damage shows up in the inelastic calculation. This is where the stress is modified for damage at the end of the current stress calculation:
https://github.com/idaholab/moose/blob/ed69223cfe138f1147bf4974200e4456fa9bf4f1/modules/tensor_mechanics/src/materials/ComputeMultipleInelasticStressBase.C#L213
This is where we use the undamaged stress from the old step:
https://github.com/idaholab/moose/blob/ed69223cfe138f1147bf4974200e4456fa9bf4f1/modules/tensor_mechanics/src/materials/ComputeMultipleInelasticStress.C#L84

**Reply by deyantian18** [reply]
Hi bwspenc,

Thank you very much. I need a bit of help understanding something in the ComputeMultipleInelasticStress module. Before the damage model starts, we have only elasticity and viscoelasticity. I've noticed that the stiffness (secant modulus=stress/strain) output shows some degradation over time because of viscoelasticity. Could you explain why this happens?

Also, '_The damage model only has the effect of decreasing the stress computed from the elastic strain_' here the elastic strain is mean the pure elasticity not including the spring's elasticity in the creep model?  just use Young's modulus instead?


Thanks a lot for your help with this.

Best,
Deyan

**Reply by bwspenc** [reply]
Yes, you're right that there should be an apparent decrease in stiffness due to viscoelasticity even before you consider damage. That is because the accumulation of inelastic strain makes the material less stiff than it would be if it were purely elastic.

What I meant in that statement about the stress computed from the elastic strain is that all the inelastic models compute elastic and inelastic strains. The stress after computing inelastic response, but before damage, is computed as C * ε_e. That elastic strain is reduced from what it would be in a purely elastic model because of the inelastic strain. The stiffness of the springs in the units of the Kelvin Voigt model will affect the inelastic strain that you get from the creep model.

**Reply by deyantian18** [reply]
Hello bwspenc,

is that okay if I use this math formulation to represent the stiffness degradation process for elastic model and inelastic model when there is a damage model ?
![image](https://github.com/idaholab/moose/assets/75801447/165efa90-ca3f-40e2-ab9a-67b39fb13467)

many thanks,
Deyan

**Reply by bwspenc** [reply]
I assume those Es are the constants for the K-V model, right? It might be correct depending on how you use them in the rest of the formulation. I think I would personally be inclined to use the undamaged versions of them in the formulation, and then use the equation at the top of your list to show how the stress is modified for damage.

**Reply by bwspenc** [reply, marked as answer]
You might find this report helpful for its description of the K-V model: https://doi.org/10.2172/1460224
It doesn't touch on how to tie that with damage, though.

**Comment by deyantian18** [comment]
Thank you very much, Ben Spencer, 

I read this file you provided, it is very helpful to understand the creep implementation in MOOSE.

I still have a problem with the stiffness tensor calculation from the file if combined with damage model,
![image](https://github.com/idaholab/moose/assets/75801447/497f4adb-e469-4ade-bf3a-aaac212d7cef)
![image](https://github.com/idaholab/moose/assets/75801447/c437f0ba-af65-4002-b283-b571b3277bca)

the C_eq is an apparent stiffness tensor associated with C_0 and C_i for first spring of the chain and i spring in the tensor. 
The C_E is the true elasticity tensor of the material which equal to the the first spring in the chain (eq9):
![image](https://github.com/idaholab/moose/assets/75801447/6a10901f-5fc0-4a23-b421-5af4ee34f5bd)

the C_eq ^(n+1) is time-dependent parameter, which results in changing with accumulation of inelastic strain in creep. If we add the damage model  with decreasing the stress: σ = (1-d) * C * (ε_total - ε_ine - ε_eigen), there the C (= C_eq ) is changing with the time? but affect independently only on stress, even the stressupdate is about to calculate the inelastic strain related.

Many thanks, 
Deyan
